AutomatedTesting/Gem/PythonTests/Prefab/tests/detach_prefab/DetachPrefab_UnderAnotherPrefab.py

```diff
@@ -45,6 +48,19 @@ async def run_test():
         # Detaches the wheel prefab instance
         Prefab.detach_prefab(wheel)
 
+        # Test undo/redo on prefab detach
+        general.undo()
+        wait_for_propagation()
+        is_prefab = editor.EditorComponentAPIBus(bus.Broadcast, "HasComponentOfType", wheel.container_entity.id,
```

Comment:
same comment as create_prefab. Instead of simply checking whether prefab container id exists or not, might be better to check the state of the children as well. For detach, you can check that the detached container entity should have the same children as the prefab and vice versa. You'll again have to find entities by name since that ids will change.

Reply:
Added additional checks for entity hierarchy.